File: lib/languages/go/assignments.go
package main

import "fmt"

func main() {
    // Variable declaration with initialization
    var name string = "Gopher"

    // Type inference
    var age = 5 // Type inferred as int

    // Short variable declaration (only inside functions)
    city := "Go City" // Shorthand for var city = "Go City"

    // Multiple variable declaration
    var x, y, z int = 1, 2, 3

    // Multiple variable declaration with type inference
    var a, b, c = 1, true, "hello"

    // Multiple short variable declarations
    i, j := 10, "value"

    // Reassignment
    i = 20 // Changing value

    // Multiple assignment
    i, j = 30, "new value"

    // Swap values
    i, j = j, i // j is now 30, i is now "new value"

    // Blank identifier for unused values
    _, value := returnTwoValues()

    fmt.Println(name, age, city, x, y, z, a, b, c, i, j, value)
}

func returnTwoValues() (int, string) {
    return 1, "value"
}

File: lib/languages/kotlin/assignments.kt
// Variable declaration with val (immutable)
val name = "Alice"  // Type inferred as String
val age: Int = 30   // Explicit type

// Variable declaration with var (mutable)
var counter = 0
counter++  // OK, counter is mutable

// Reassignment
// name = "Bob"  // Error: val cannot be reassigned
counter = 42      // OK, var can be reassigned

// Destructuring assignment
val (x, y) = Pair(1, 2)
val (key, value) = "key" to "value"

// Destructuring from data classes
data class Person(val name: String, val age: Int)
val person = Person("Alice", 30)
val (personName, personAge) = person

// Destructuring in lambda parameters
val map = mapOf("a" to 1, "b" to 2)
map.forEach { (k, v) -> println("$k -> $v") }

// Late initialization
lateinit var lateString: String
// Use lateString later after initialization

// Lazy initialization
val lazyValue: String by lazy {
   println("Computed!")
   "Hello"
}
// lazyValue is computed on first access

// Delegated properties
import kotlin.properties.Delegates
var observed: Int by Delegates.observable(0) { _, old, new ->
   println("Changed from $old to $new")
}

